start-from-zero/Javascript/src/views/new-evaluation/new-evaluation.js

```diff
@@ -1,150 +1,178 @@
+        xhr.onreadystatechange = function(){
+            if(this.readyState === 4){
+                if(this.status == 200){
+                    try{
+                        let response = JSON.parse(xhr.responseText);
+                        windowObjectWrap.NewEvaluationPage = function(options) {
+                            this.render = `
+                            ${windowObjectWrap.NAV()}
+                            <section class="form-candidate">
+                                <div class="container">
+                                    <div class="row">
+                                        <div class="col-md-12">
+                                            <form action="" class="user-input" id='form-new' method='post'>
+                                                ${Inputs(
+                                                    response.inputPlaceholder
+                                                )}
+                                                ${TechnicalLevelPicker(
+                                                    response.technicalData
+                                                )}
+                                                ${TextArea(
+                                                    response.textarea
+                                                )}
+                                                ${SelectOption(
+                                                    response.fieldset
+                                                )}
+                                                <div class="button-wrap">
+                                                    <input type="button" id='btn' class='submit-button' value='Submit'>
+                                                </div>
+                                                
+                                            </form>
+                                        </div>
+                                    </div>
+                                </div>
+                            </section>
+                            ${windowObjectWrap.Footer()}
+                            `
+                            ;
+                        }
                         
-                    </form>
-                </div>
-            </div>
-        </div>
-    </section>
-    ${windowObjectWrap.Footer()}
-    `
-    
-    ;
+                        
+                        const TableData = function (options = {}) {
+                            const currentLevel = options.headers.map((el) => `
+                                    <td class='technical-level-td'>${el}</td>
+                                `).join('');
+                            return currentLevel;
+                        };
+                        
+                        const TableCheckbox = function (options = {}) {
+                            const checkboxArr = [];
+                            options.columnData.forEach((d) => {
```

Comment:
Not that this is necessarily wrong, but there are other, better suited, APIs for iterating over collections and mapping their elements to corresponding pieces of information.